src/app/reporting/Engine.cpp

```diff
@@ -676,14 +674,14 @@ void Engine::Run()
     // This isn't strictly necessary, but does make it easier to debug issues in this code if they
     // do arise.
     //
-    if (mCurReadHandlerIdx >= mpImEngine->mReadHandlers.Allocated())
+    if (mCurReadHandlerIdx >= mDelegate->GetReadHandlerPool().Allocated())
     {
         mCurReadHandlerIdx = 0;
     }
 
     bool allReadClean = true;
 
-    mpImEngine->mReadHandlers.ForEachActiveObject([&allReadClean](ReadHandler * handler) {
+    mDelegate->GetReadHandlerPool().ForEachActiveObject([&allReadClean](ReadHandler * handler) {
```

Comment:
There should be a ForEachReadHandler on the delegate or something.

Reply:
Will have to see ... this is rough because we capture here and passing in std::function requires heap (not sure if this requires heap or not ... but probably not).

This neeeds a bit of thought. Overall I woner if this should be some `DoFoo` request inside the Delegate and then move the inside logic into InteractionModelEngine. I will iterate a bit here.